### PR TITLE
Weather API

### PR DESCRIPTION
**Pull request by AymanJabr**
In this branch, I created a very simple weather app, that lets you input the name of a city, and it will get you accurate information about the current weather condition in that city, for example: temperature, pressure, cloud conditions...

### CODE REVIEW

**Comment by AymanJabr** [comment]
# The app will not work unless you supply it with a weather app API key

**Comment by MuriloRoque** [comment]
## Status: Invalid :red_circle: 

### Hello @AymanJabr :wave: 

### Unfortunately I need to invalidate your submission because of the following issues:

- [x] Your Pull Request is not checking for linter errors, please configure Github Actions.

Kindly make the requested changes and submit them for another review, please :pray: 

I will be on Slack 24/7 in case you have any questions about this review :handshake: 

Murilo Roque Paiva da Silva

- Github: [@muriloroque](https://github.com/MuriloRoque)
- Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)

[Twitter Follow](https://twitter.com/MuriloRoquePai1)

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Changes Required 🙏🏿 🙏🏿 🙏🏿 
Nice work on this PR.
### Features implemented 🍾 🍾 🍾 
- weather information is displayed for a given city

### Changes required 🌵 🌵 🌵 
- change the background image of the app according to weather condition
- add a toggle button to change between Celsius and Fahrenheit values
- add a search bar for searching different cities. Right now a prompt alert is used to input weather location. Please use a real search widget.
- update the `Getting started` section with relevant information. Add explicit commands for cloning, changing directory, installing npm packages, etc
- add error handling for `promises`. This is the part where you are making API calls
- use webpack in your app.
- all webpack related packages should be included in `package.json` like `webpack`, `webpack-cli`, `webpack-dev-server` etc. Also, those packages should be added as **devDependencies**
- add these packages to the scripts section of the package.json file:
```json
"scripts": {
    "dev": "webpack --mode development",
    "build": "webpack --mode production",
    "start": "webpack-dev-server --mode development --open" // optional
  },
```

### Martin Afani
#### Slack: whiz25

**Comment by AymanJabr** [comment]
## No reason to Add Webpack to this project, and it is not required

**Review by Temmarie — CHANGES_REQUESTED** [review]
## Changes Required! :recycle:

### Hi @AymanJabr     ! :wave: :wave:

#### Good job implementing some of the requested changes.  :clap: However, there are changes required for you to make for this project to be approved. Kindly work on them and submit them for another code review.
- - -

### **Recommended**

- [x] It uses Webpack ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/webpack_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/)).
- [x] All webpack related packages should be included in `package.json` like `webpack`, `webpack-cli`, `webpack-dev-server` etc. Also, those packages should be added as **devDependencies** ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/devdep_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/))
- Your project is currently missing the `package.json` file.
- [x] Add a search bar for searching different cities. Right now a prompt alert is used to input weather location. Please use a real search widget.



__Kindly make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink:
__Grace Tamara Ekunola __
[GitHub](https://github.com/temmarie) | [LinkedIn](https://www.linkedin.com/in/ekunola-grace/) | [Twitter](https://twitter.com/TemmarieW)

File: index.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    documentLocation.innerHTML += currentLocation;
+    documentLatitude.innerHTML += coordinates[1];
+    documentLongitude.innerHTML += coordinates[0];
+    documentTemperature.innerHTML += avgTemp;
+    documentPressure.innerHTML += pressure;
+    documentHumidity.innerHTML += humidity;
+    documentDescription.innerHTML += description;
+    documentIcon.src = `http://openweathermap.org/img/w/${icon}.png`;
+  }).catch((e) => console.error(e));
```
Comment (rindrajosia) [inline index.js]: [REQUIRED]No console.error commited to repo.

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Changes Required :red_circle:
To Highlight: :angel:

- Good job by adding webpack :+1: 


To improve :factory_worker: [REQUIRED]:

- [x] I suggest that you use tools like grammarly in order to help you with your grammar in the readme.
- [x] No `console` commited to repo. It's ugly, it kills performance. and it can make confidential data be visible in the browser tools. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/console_log_requirement.png))
- [x] Keep the application `logic separated from DOM manipulation tasks `([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/dom_logic_separation_requirement.png))
- [x]  I got error when I enter a location that doesn't exist. Please, make sure that your APP has `error handling for promises `([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/error_handling_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/))


- [x] Your APP doesn't open the `index.html` file when I use command `npm run start` I suggest that you move your `index.html `file in the `dist folder`.




Nice to have:

- [ ] it has a ‘`loading`’ component that displays from the time the form is submitted until the information comes back from the API ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/loading_component_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/weather-app))

That's it, you can submit another code review request after making these changes :recycle: :innocent:
With all my respect :pray:
Rindra Josia

**Review by rindrajosia — APPROVED** [review]
Approved! :tada:
Congratulations are in order, @Ayman! :clap: Keep up the good work!

[REQUIRED] 

- [ ] Modify your readme based on your last update. For example: this `Insert it at line 1, in the index.js where it tells you to do so` should be` Insert it at line 9, in the index.js where it tells you to do so`.

[OPTIONAL]
- [ ] It will be nice if you put all JS files in **scr** folder
- [ ]  it fetches the current geolocation of the user when the app is first accessed and fetches the weather from that location

All the best on your next projects! :wink:

**Pull request merged**